### ISSUE TITLE
Fails to compile on clang with old c++ standard

### ISSUE THREAD

**Issue by alastair** [MTG/essentia#795]
Clang defaults to c++98, which seems to be too old for some features that are being used. Specifically, it fails at

```
../src/algorithms/standard/constantq.cpp:103:58: error: implicit conversion from '_Complex double' to 'double' is not permitted in C++
  vector<complex<double> > hammingWindow(_FFTLength, 0.0 + 0.0i);
                                                     ~~~~^~~~~~
```

This doesn't fail on g++ 8.2, which defaults to gnu++14.

It'd be good to choose a specific standard and add it to `CXXFLAGS`

**Comment by alastair** [comment]
Workaround is to set the c++ standard while compiling:

    CXXFLAGS=-std=c++14 ./waf configure

**Comment by dbogdanov** [comment]
Which version of clang is that? I suppose since clang 6.0 it [defaults to c++14](https://www.phoronix.com/scan.php?page=news_item&px=LLVM-Clang-60-CPP14).

**Comment by alastair** [comment]
unclear, apple's clang reports the version of xcode, not clang:
```
$ clang -v
Apple LLVM version 10.0.0 (clang-1000.10.44.2)
```

**Comment by alastair** [comment]
Fixed by #804
